woocommerce/google-listings-and-ads
# [API Pull] Tweak Variations Notifications

### Changes proposed in this Pull Request:

This PR tweaks the logic for notifications when handling variations. In particular:

- FIXED: We were sending create notifications for variations when the parent was not published still
- FIXED: We were only sending delete notifications for the parent when a variation was deleted.
- FIXED: SOme unhandled exceptions when we try to access. a product that is not in the database anymore. 

### Detailed test instructions:


1. Checkout PR
2. Test that creation/update/deletion of simple product works as always
3. Create a Variable Product and generate variations for It 
4. Go to Woo - Status - Scheduled actions - That step should not schedule any notifications
5. Asign price to some of the variations and publish the product
6. Go to Woo - Status - Scheduled actions - See scheduled notifications create for the variations with price + the variable product
7. Run the actions and see if they are logged in the gla log under Woo - Status - logs
8. Edit one of the variations without price
9. It doesn't schedule or log anything as it was not created yet
10. Remove one of the variations without price
11.  It doesn't schedule or log anything as it was not created yet
12. Edit one of the variations with price 
13. Go to Woo - Status - Scheduled actions - See scheduled notifications update for the variations with price + the variable product. Run them
14. Remove the variation with price 
15. Go to Woo - Status - Scheduled actions - See scheduled notifications update for the variations with price + the variable product and one delete notification in the logs (directly without schedule) for the one you deleted with offer_id
16. Set the variable product as DONT SYNC AND SHOW
17.  Go to Woo - Status - Scheduled actions - See scheduled notifications delete for the variations with price + the variable product 
18. Before running it. Go back and set the variable product as  SYNC AND SHOW
19. Go to Woo - Status - Scheduled actions - See scheduled notifications update for the variations with price + the variable product 
20. Run the delete notifications of the step 17
21. See in the logs they are not logged (a filter prevented it, because we set is as visible in step 18)
22. Run the tasks from step 19
23. See them logged in the log
24. Try to delete products and variations in different ways to see if everything is handled properly.


### Additional details:


### Changelog entry

>
